fe/fe-core/src/main/java/com/starrocks/sql/plan/PlanFragmentBuilder.java

```diff
@@ -737,6 +738,9 @@ public PlanFragment visitPhysicalOlapScan(OptExpression optExpr, ExecPlan contex
                 context.getColRefToExpr().put(entry.getKey(), new SlotRef(entry.getKey().toString(), slotDescriptor));
             }
 
+            // set column access path
+            scanNode.setColumnAccessPaths(node.getColumnAccessPaths());
```

Comment:
I think maybe you need to remove below code in OlapScan, becuase you are using ColumnAccessPath instead.
```sql
if (slotDescriptor.getOriginType().isComplexType()) {
    slotDescriptor.setOriginType(entry.getKey().getType());
    slotDescriptor.setType(entry.getKey().getType());
}
```

Reply:
I was forbidden PruneComplexTypeRule work on OlapScan